core/pkg/ingress/types.go

```diff
@@ -134,6 +135,8 @@ type Backend struct {
 	Secure bool `json:"secure"`
 	// Endpoints contains the list of endpoints currently running
 	Endpoints []Endpoint `json:"endpoints"`
+	// StickySession contains the StickyConfig object with stickness configuration
+	StickySession stickysession.StickyConfig `json:"stickysession,omitempty"`
```

Comment:
How about something like
```go
Backend {
  SessionAffinity SessionAffinityConfig
}

// SessionAffinityConfig describes different affinity configurations for new sessions. 
// Once a session is mapped to a backend based on some affinity setting, it 
// retains that mapping till the backend goes down, or the ingress controller 
// restarts. Exactly one of these values will be set on the upstream, since multiple
// affinity values are incompatible. Once set, the backend makes no guarantees 
// about honoring updates. 
type SessionAffinityConfig {
  *CookieSessionAffinity 
  // *ClientSessionAffinity -- can add this later
}

Type CookieSessionAffinity {
  CookieName string
  HashAlgorithm
  // can't we assume the non nil pointer as proof that it needs to be enabled? 
}
```
Thoughts? I am open to other forms of refactor that make handling affinity more generic, because it is more generic than just sticky sessions. In fact one can even express "nodeaffinity" as "loadbalance to an endpoint on the same node, if one exists", and by extension define a zone/region affinity metrics to optimize egress costs on clouds.